examples/pastebin/src/tests/mod.rs

```diff
@@ -0,0 +1,69 @@
+use std::io::prelude::*;
+use std::fs::File;
+use std::path::Path;
+
+use super::rocket;
+use rocket::testing::MockRequest;
+use rocket::http::{Method, Status};
+
+#[test]
+fn index() {
+    let rocket = rocket::ignite().mount("/", routes![super::index]);
+    let mut req = MockRequest::new(Method::Get, "/");
+    let mut response = req.dispatch_with(&rocket);
+
+    assert_eq!(response.status(), Status::Ok);
+
+    let body_str = response.body().and_then(|body| body.into_string());
+    assert_eq!(body_str, Some(include_str!("usage.txt").to_string()));
+}
+
+#[test]
+fn upload() {
+    let rocket = rocket::ignite().mount("/", routes![super::upload]);
+    let mut req = MockRequest::new(Method::Post, "/")
+        .body(include_str!("file.txt").to_string());
+    let mut response = req.dispatch_with(&rocket);
+
+    assert_eq!(response.status(), Status::Ok);
+
+    match response.body().and_then(|body| body.into_string()) {
+        Some(url) => {
+            match url.trim().split("/").last() {
+                Some(paste_id) => {
+                    let filename = format!("upload/{}", paste_id);
+                    assert!(Path::new(&filename).exists());
```

Comment:
This is assuming a particular implementation, which I don't think these tests should do. I think instead we should combine this with the next test and do a `POST` followed by a `GET`. Would be cool to do several of them, in fact, to make sure everything looks good.